### ISSUE TITLE
Tracking python lists (or ListWrapper) in keras custom preprocessing layers

### ISSUE THREAD

**Issue by ianqouda** [tensorflow/tensorflow#47910]
### System information

-   **Custom code**
-   **OS Platform and Distribution**:

> - PRETTY_NAME="Debian GNU/Linux 9 (stretch)"
> - NAME="Debian GNU/Linux"
> - VERSION_ID="9"
> - VERSION="9 (stretch)"
> - ID=debian

-   **TensorFlow installed from binaries**:
-   **TensorFlow version**: 2.4.0
-   **Python version**: 3.6.8

### Problem
Hi,
My problem is the following: I try to create a custom keras preprocessing layer in which I absolutely need an attribute which is a regular python list computed via adapt method. When I include my layer in model and I save it (via SavedModel), my list attribute becomes empty at load time. Is there anyway to make this list (or any other python object) trackable so that it is properly loaded?

Some additional points:
- It can be made trackable if I use a variable instead of a list but in my case a list is necessary.
- I still can use a variable and transform it to a list at call time, but this will work only in eager mode and I need it to work at graph mode also

### Source code / logs
Here a simple example to reproduce my issue:
```python
import numpy as np
import tensorflow as tf

from tensorflow.python.ops import math_ops


@tf.keras.utils.register_keras_serializable()
class BucketizeLayer(tf.keras.layers.experimental.preprocessing.PreprocessingLayer):
    
    def __init__(self, quantiles, **kwargs):
        self.quantiles = quantiles
        super(BucketizeLayer, self).__init__(**kwargs)
        self._boundaries = None
        
    def adapt(self, data):
        if isinstance(data, tf.Tensor):
            data = data.numpy()
            
        self._boundaries = np.nanquantile(data, self.quantiles).tolist()
        
    def call(self, data):
        return math_ops.bucketize(data, self._boundaries)
    
    def get_config(self):
        config = {'quantiles': self.quantiles}
        base_config = super(BucketizeLayer, self).get_config()
        return dict(**config, **base_config)


data = 3 * np.random.randn(100) + 5
bucketize = BucketizeLayer(quantiles=[0.1, 0.5, 0.9])
bucketize.adapt(data)
print(bucketize(data))


inp = tf.keras.Input(shape=(1,), dtype=tf.float32)
model = tf.keras.models.Model(inp, bucketize(inp))
model.save('bucketize/')
del model
model = tf.keras.models.load_model('bucketize/')
model.layers[1]._boundaries
# output: ListWrapper([])
```

**Comment by Saduf2019** [comment]
@ianqouda 
i ran the code shared and do not see an empty list, please refer to the [gist here](https://colab.research.google.com/gist/Saduf2019/ac2b96a3d295f7744050b242d95fffb4/untitled565.ipynb) and share a colab gist with the error reported.

**Comment by ianqouda** [comment]
@Saduf2019 thank you for your reponse, here is a [https://colab.research.google.com/gist/Saduf2019/ac2b96a3d295f7744050b242d95fffb4/untitled565.ipynb#scrollTo=DlvaRQRQr97I](gist) with more detailed code. The problem is that the attribute '_boubdaries' becomes empty after loading the model.

**Comment by Saduf2019** [comment]
I am able to replicate the issue reported on tf 2.4 and nightly, please find the [gist here](https://colab.research.google.com/gist/Saduf2019/d0a9b04b87978907c7c112f7fad16104/untitled580.ipynb)

**Comment by jvishnuvardhan** [comment]
@ianqouda I updated your code (mainly adding `get_config` method and `self._boundaries = boundaries`). With those modifications, everything works as expected. Please check the [gist here](https://colab.research.google.com/gist/jvishnuvardhan/23725bbc9fb3d13952172dc1fb77a229/untitled580.ipynb).

Please verify once and close the issue if this was resolved for you. Thanks!

**Comment by ianqouda** [comment]
@jvishnuvardhan Thank you very much! This works fine for this example I showed here but in more complexe context (which is my case) this solution alters a little bit the behaviour of my layer. More precisely if I have this boundary as an argument which I can set manually and force the adapt method not to override it then this solution would not work any more.

In the mean time I found another way to do it by creating a trackable list through inhereting from this object from tensorflow.train.experimental.PythonState and it works fine!
